app/frontend/src/features/communities/NewPageForm.tsx

```diff
@@ -22,53 +22,53 @@ type NewPageInputs = {
 };
 
 export default function CompleteSignup() {
-  const dispatch = useAppDispatch();
-
   const {
     control,
     register,
     handleSubmit,
     setValue,
     errors,
-    getValues,
   } = useForm<NewPageInputs>({
     shouldUnregister: false,
     mode: "onBlur",
   });
 
-  const [loading, setLoading] = useState(false);
-
-  const location = useLocation();
   const history = useHistory();
 
   const [alertState, setShowAlertState] = useState<
     "success" | "error" | undefined
   >();
   const [errorMessage, setErrorMessage] = useState<string | null>(null);
 
-  const submitForm = handleSubmit(async (data: NewPageInputs) => {
-    try {
-      const page = unwrapResult(await dispatch(createPage(data))) as Page.AsObject;
+  const {
+    mutate: createPage,
+    isLoading: isCreateLoading,
+  } = useMutation<Page.AsObject, GrpcError, NewPageInputs>(
+    ({title, content, address, lat, lng}: NewPageInputs) => service.pages.createPage(title, content, address, lat, lng),
+    {
+      onSuccess: (page) => {
       setShowAlertState("success");
       history.push(`${pageRoute}/${page.pageId}/${page.slug}`);
-    } catch (error) {
-      console.error(error)
-      setShowAlertState("error");
-      setErrorMessage(error.message);
+    },
+      onError: (error) => {
+        console.error(error)
+        setShowAlertState("error");
+        setErrorMessage(error.message);
+      }
     }
-  });
+  );
+
+  const onSubmit = handleSubmit((data: NewPageInputs) =>
+    createPage(data)
+  );
 
   return (
     <>
-      {alertState === "success" ? (
-        <Alert severity={alertState}>Successfully created page!</Alert>
-      ) : alertState === "error" ? (
-        <Alert severity={alertState}>{errorMessage}</Alert>
-      ) : null}
-      {loading ? (
+      {alertState && <Alert severity={alertState}>{errorMessage || "Successfully created page!"}</Alert>}
```

Comment:
```suggestion
      {error && <Alert severity="error"}>{error?.message}</Alert>}
```
See other comment. I don't think you need to show success state since it redirects straight away.

Reply:
OK, yeah. That's why I had the error state.